### ISSUE TITLE
TriStateWindow AddOn - is this a binding?

### ISSUE THREAD

**Issue by hotzen** [openhab/openhab2-addons#2392]
this is not a bug, just need to know the general direction.

Status Quo:
- Contact is OPEN on TRUE and Closed on FALSE 
- Windows with 3 states OPEN, CLOSED, AJAR need to define 2 Contacts for OPEN/CLOSED and AJAR/CLOSED
- To have the 3 states in one Item, you need to define a rule for EACH window

I want to provide an Addon that allows a simple String or Number-Item to be bound to the Contact-Items, having the state OPEN, CLOSED, AJAR or 1,0,2 respectively.
The addon also needs to (optionally) negate the Contact's logic because windows are usually CLOSED when the Contact is TRUE (in contrast to the current hardcoded behaviour where (TRUE means Contact-Open).

Should I implement this as a binding or is there a better suited OpenHAB/ESH Concept?
Basically it's just a parametrized rule: 
myTriStateWindowItem.pushUpdate( evalTriStateWindow(OpenContact, AjarContact) )

**Comment by wborn** [comment]
Did you already find an answer to this question @hotzen ?

**Comment by davidgraeff** [comment]
openHAB profiles, transformations and rules are the way to go here.
